proxy/agent_test.go

```diff
@@ -364,3 +364,41 @@ func TestCascadingAbort(t *testing.T) {
 	a.ScheduleConfigUpdate(2)
 	<-ctx.Done()
 }
+
+// TestLockup plays a scenario that may cause a deadlock
+//  * start epoch 0
+//  * start epoch 1 (wait till sets desired and current config)
+//  * epoch 0 crashes, triggers abort
+//  * epoch 1 aborts
+//  * progress should be made
+func TestLockup(t *testing.T) {
```

Comment:
does the test fail without the fix ?

Reply:
Yes

Reply:
Aborts are triggered by an abnormal exit code.

Reply:
what trigger those ?

Reply:
Any malfunction in envoy. Exit on draining is not abnormal. The reason for aborts is that we found out that once envoy starts crashing, the regular draining protocol breaks down, so we have to kill all active proxies (draining and new ones).

Reply:
once detected, would it be easier to restart the agent too (and kill all child before that) or let kube restart the container maybe ?

Reply:
Yes, but that would also restart application container.